### PR TITLE
Switch Travis to use tox

### PR DESCRIPTION
**Pull request by sigmavirus24**
This reduces confusion about how travis runs tests versus a local
developer.

**Note** this also adds a Flake8 test env and runs it at the gate to catch other instances of https://github.com/PyCQA/pycodestyle/pull/540

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -14,3 +14,10 @@ commands =
 	{envpython} pycodestyle.py --statistics pycodestyle.py
 	{envpython} pycodestyle.py --doctest
 	{envpython} -m testsuite.test_all
+
+
+[testenv:flake8]
+deps =
+    flake8
+commands =
+    flake8 pycodestyle.py
```
Comment (IanLee1521) [inline tox.ini]: What's the rationale behind running flake8 on pycodestyle? Not specifically disagreeing, just seems to not be part of the "Switch Travis to use tox" issue.

File: .travis.yml [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
-  - pip install -e .
+  - pip install tox
   - pip list
-script:
-  - python pycodestyle.py --testsuite testsuite
-  - python pycodestyle.py --statistics pycodestyle.py
-  - python pycodestyle.py --doctest
-  - python setup.py test
+script: tox
+matrix:
```
Comment (IanLee1521) [inline .travis.yml]: Why convert from the python to the matrix version of this?

File: .travis.yml [hunk 3]
```diff
@@ -1,23 +1,29 @@
...
 install:
-  - pip install -e .
+  - pip install tox
   - pip list
-script:
-  - python pycodestyle.py --testsuite testsuite
-  - python pycodestyle.py --statistics pycodestyle.py
-  - python pycodestyle.py --doctest
-  - python setup.py test
+script: tox
```
Comment (IanLee1521) [inline .travis.yml]: I like this part! I had come up with the same idea and forgotten to change it... Great minds! :)

File: .travis.yml (hunk 2 above)
Comment (jayvdb) [inline .travis.yml]: It can be avoided using `tox-travis`

Comment (sigmavirus24) [inline .travis.yml]: @IanLee1521 it matches the usage of the majority of projects that have test environments that are more than just "test on these versions of python". It would allow the project to, for example, also run a tox test environment to ensure that the documentation passes a tool like doc8 and builds in sphinx with all warnings treated as errors, etc. (Effectively testing that your docs meet a certain style and actually build.)

Given how many more projects use the matrix approach rather than using the rather young tox-travis tool (which I've not seen any project of any size use), I think this is going to be the better way to manage our usage of tox within travis.

File: tox.ini (hunk 1 above)
Comment (sigmavirus24) [inline tox.ini]: Flake8, as I'm sure you know, includes pyflakes. When I was writing some of the more recent pull requests, I found several pyflakes lint problems (which I also submitted separate pull requests for). If pycodestyle runs flake8 on all submitted changes, then those shouldn't pop up again.

It's something of an extra check to ensure the quality of pycodestyle doesn't dip.
